code/datums/voice_of_god_command.dm

```diff
@@ -0,0 +1,427 @@
+			power_multiplier *= 1.4
+		//Why are you speaking
+		if(user.mind.assigned_role == "Mime")
+			power_multiplier *= 0.5
+
+	//Cultists are closer to their gods and are more powerful, but they'll give themselves away
+	if(iscultist(user))
+		power_multiplier *= 2
+
+	//Try to check if the speaker specified a name or a job to focus on
+	var/list/specific_listeners = list()
+	var/found_string = null
+
+	//Get the proper job titles
+	message = get_full_job_name(message)
+
+	for(var/V in listeners)
+		var/mob/living/L = V
+		if(findtext(message, L.real_name, 1, length(L.real_name) + 1))
+			specific_listeners += L //focus on those with the specified name
+			//Cut out the name so it doesn't trigger commands
+			found_string = L.real_name
+
+		else if(findtext(message, L.first_name(), 1, length(L.first_name()) + 1))
+			specific_listeners += L //focus on those with the specified name
+			//Cut out the name so it doesn't trigger commands
+			found_string = L.first_name()
+
+		else if(L.mind && L.mind.assigned_role && findtext(message, L.mind.assigned_role, 1, length(L.mind.assigned_role) + 1))
+			specific_listeners += L //focus on those with the specified job
+			//Cut out the job so it doesn't trigger commands
+			found_string = L.mind.assigned_role
+
+	if(specific_listeners.len)
+		listeners = specific_listeners
+		power_multiplier *= (1 + (1/specific_listeners.len)) //2x on a single guy, 1.5x on two and so on
+		message = copytext(message, length(found_string) + 1)
+
+	for(var/datum/voice_of_god_command/command as anything in GLOB.voice_of_god_commands)
+		if(findtext(message, command.trigger))
+			. = command.execute(listeners, user, power_multiplier, message)
+
+	if(message_admins)
+		message_admins("[ADMIN_LOOKUPFLW(user)] has said '[log_message]' with a Voice of God, affecting [english_list(listeners)], with a power multiplier of [power_multiplier].")
+	log_game("[key_name(user)] has said '[log_message]' with a Voice of God, affecting [english_list(listeners)], with a power multiplier of [power_multiplier].")
+	SSblackbox.record_feedback("tally", "voice_of_god", 1, log_message)
+
+/// Voice of god command datums that are used in [/proc/voice_of_god()]
+/datum/voice_of_god_command
+	///a text string or regex that triggers the command.
+	var/trigger
+	/// Is the trigger supposed to be a regex? If so, convert it to such on New()
+	var/is_regex = FALSE
+	/// cooldown variable which is then returned by [proc/voice_of_god]
+	var/cooldown = COOLDOWN_MEME
+
+/datum/voice_of_god_command/New()
+	if(is_regex)
+		trigger = regex(trigger)
+
+/*
+ * What happens when the command is triggered.
+ * Args:
+ * * listeners: the list of living mobs who are affected by the command.
+ * * user: the one who casted Voice of God
+ * * power_multiplier: multiplies the power of the command, most times.
+ */
+/datum/voice_of_god_command/proc/execute(list/listeners = list(), mob/living/user, power_multiplier = 1, message)
```

Comment:
You shouldn't be doing `= list()` everywhere, you should just have your code work with null, or know that it'll never be null.

`for (var/name in null)` passes fine.

Reply:
Same with everywhere else.